test-integ/peering_commontopo/setup.go

```diff
@@ -0,0 +1,58 @@
+package peering
+
+import (
+	"flag"
+	"testing"
+)
+
+// Test that use commonTopo should implement commonTopoSuite.
+//
+// Tests that use commonTopo are either cooperative or non-cooperative. Non-cooperative
+// uses of commonTopo include is anything that may interfere with other tests, namely
+// mutations, such as:
+// - any calls to commonTopo.Relaunch; this is generally disruptive to other tests
+// - stopping or disabling nodes
+// - ...
+//
+// Cooperative tests should just call testFuncMayReuseCommonTopo() to ensure they
+// are run in the correct `sharetopo` mode. They should also ensure they are included
+// in the commonTopoSuites slice in TestSuitesOnSharedTopo.
+type commonTopoSuite interface {
+	testName() string
+	setup(*testing.T, *commonTopo)
+	test(*testing.T, *commonTopo)
+}
+
+var commonTopologyFlag = flag.Bool("commontopo", false, "run tests with common topology")
+
+func commonTopologyTest(t *testing.T) {
+	t.Helper()
+	if *commonTopologyFlag {
+		t.Skip(`Skipping: duplicate test run. See "TestSuitesOnSharedTopo"`)
```

Comment:
nit: The word "Skipping" is redundant here
<img width="571" alt="image" src="https://github.com/hashicorp/consul/assets/115657443/1c801935-da6f-4c62-94b5-ae25b1fb498f">